server/recceiver/cfstore.py

```diff
@@ -20,37 +19,249 @@
 from twisted.internet.threads import deferToThread
 
 from . import interfaces
+from .interfaces import CommitTransaction
+from .processors import ConfigAdapter
 
 _log = logging.getLogger(__name__)
 
-# ITRANSACTION FORMAT:
-#
-# source_address = source address
-# records_to_add = records ein added ( recname, rectype, {key:val})
-# records_to_delete = a set() of records which are being removed
-# client_infos = dictionary of client client_infos
-# record_infos_to_add = additional client_infos being added to existing records
-# "recid: {key:value}"
-#
-
 __all__ = ["CFProcessor"]
 
-RECCEIVERID_KEY = "recceiverID"
 RECCEIVERID_DEFAULT = socket.gethostname()
+DEFAULT_MAX_CHANNEL_NAME_QUERY_LENGTH = 600
+DEFAULT_QUERY_LIMIT = 10_000
+
+
+class PVStatus(enum.Enum):
+    """PV Status values."""
+
+    Active = enum.auto()
+    Inactive = enum.auto()
+
+
+@dataclass
+class CFConfig:
+    """Configuration options for the CF Processor"""
+
+    alias_enabled: bool = False
+    record_type_enabled: bool = False
+    environment_variables: str = ""
+    info_tags: str = ""
+    ioc_connection_info: bool = True
+    record_description_enabled: bool = False
+    clean_on_start: bool = True
+    clean_on_stop: bool = True
+    username: str = "cfstore"
+    recceiver_id: str = RECCEIVERID_DEFAULT
+    timezone: Optional[str] = None
+    cf_query_limit: int = DEFAULT_QUERY_LIMIT
+
+    @classmethod
+    def loads(cls, conf: ConfigAdapter) -> "CFConfig":
```

Comment:
`loads` usually means "load from **s**tring" (and then `load` is typically from file), but I guess this is OK because `from_configadapter` would be a bit odd/is not immediately clear (that class is a bit strange TBH)